packages/common/container-definitions/src/audience.ts

```diff
@@ -67,4 +73,24 @@ export interface IAudience extends EventEmitter {
 	 * does not technically have a clientId tied to an active connection to the service.
 	 */
 	getMember(clientId: string): IClient | undefined;
+
+	/**
+	 * Returns this client's clientId. undefined if this client has never connected to the ordering service.
+	 * @experimental
+	 * 
+	 * @remarks
+	 * This API is experimental.
+	 *
+	 * Whenever this property changes, the "clientIdChanged" event is fired on this object.
```

Comment:
```suggestion
	 * For a container which was connected to the ordering service at some point but is currently disconnected, this property
	 * returns the clientId acquired during the most recent connection.
	 *
	 * Whenever this property changes, the "clientIdChanged" event is fired on this object.
```

Reply:
Your adjustment is not technically true (it depends on the visibility of connections). We may connect and disconnect without ever changing clientId. That's because we did not go far enough on connection sequence to change clientId and lost it. It's true only for connections that were exposed to users (i.e. we raised "connected" event).
We can also have connection, but not notify (to runtime layers) about it, with exception of clientId changes (read-only file permissions).

I'd rather prefer not to add such text, and in general, slowly remove any mentions to connections at all. Even "connected" even I'd rather see narrating about being caught-up, and not about having a connection. We can't remove completely connections (as sending signals requires knowing about connected / disconnected states).

I'm modifying this text to be the following:

	 * It's guaranteed that these events happen at the same time (synchronously, one after another):
	 * 1. "clientIdChanged" event on this object fires
	 * 2. the change of current clientId
	 * 3. current clientId is added to members of audience
	 * If  "connected" event fires, it will fire at the same time. "connected" event may not fired at some layers (like container runtime layer)
	 * in some cases (like user has read-only permissions to container).

Reply:
My concern was mostly about being explicit about what `currentClientId` means when a container is not connected to a server, because we had run into something similar using Audience in Devtools which resulted in [this doc update](https://github.com/microsoft/FluidFramework/pull/19463) (which you might want to review/update as well as part of a plan to remove the idea of the connection or connected/disconnected state, if we can go through with that). In general, we don't seem to be clear on what using IAudience is supposed to guarantee when a container is not connected. Maybe it's supposed to be a "snapshot as of the last time we were connected" (e.g. which members are/were connected to the container at that time), but whether that or something else is the case, right now we just don't say anything that I'm aware of. So trying to start making it explicit if possible.

[Related ADO item](https://dev.azure.com/fluidframework/internal/_workitems/edit/7169).

Reply:
The remark seems Ok to me. But that's slightly different topic - it's about when we clear and when we populate audience on disconnect / reconnect. And the answer is - it's not defined.
when it comes to current clientId, then behaviors are very well defined and will never change (as things will break). At least from observable POV.

I'll add this clarification that I hope removes confusion (but omits talking about connections that never were visible to user):

> It changes only when client has reconnected and caught up with latest ops.